### DISCUSSION TITLE
Texture artifacts with KTX2 with BasisU supercompression

### DISCUSSION THREAD

**Discussion by callumlinden** [zeux/meshoptimizer#743] [Q&A] [answered]
My work flow involves taking a GLB mesh with no textures from a content team, applying X sets of N meshes to it and saving the result off as a X new GLBs. 

Then I use gltfpack to optimize/compress the result.  The size differences are dramatic but even with `-tc -tq 10` I observe artifacts in the textures for the output GLB file. They're not present if texture compression is disabled entirely.

Without `-tc`:
![no_compression](https://github.com/user-attachments/assets/9de96832-c771-4506-977c-feea5b724cdb)

With `-tc -tq 10`:
![tc tq 10](https://github.com/user-attachments/assets/5af58aae-734f-45df-9c12-797cae048249)

Anything else I can try or is that just the nature of KTX2/BasisU supercompression?

gltfpack 0.21

**Comment by zeux** [comment, marked as answer]
Some amount of color artifacts is inevitable; I'm surprised it's this drastic but maybe that makes sense on some blocks like this. Could be a bug in the underlying library used (https://github.com/BinomialLLC/basis_universal) but likely just a limitation of format.

KTX2/BasisU has two algorithms, ETC1 & UASTC; I would try UASTC with a low quality setting (eg `-tu -tq 3`). It's likely to be much larger than ETC1S (which `-tc` uses), but much much much higher quality. Unfortunately there's not a great middle ground between the two atm (eg "better quality at a fairly small download size").

**Reply by callumlinden** [reply]
Understood - thank you for the detailed reply.
